src/schema/text_options.rs

```diff
@@ -16,13 +16,74 @@ pub struct TextOptions {
     #[serde(default)]
     stored: bool,
     #[serde(default)]
-    fast: bool,
+    fast: FastFieldOptions,
     #[serde(default)]
     #[serde(skip_serializing_if = "is_false")]
-    /// coerce values if they are not of type string
+    /// coerce values into string if they are not of type string
     coerce: bool,
 }
 
+#[derive(Clone, Debug, Default, PartialEq)]
+enum FastFieldOptions {
+    Enabled,
+    #[default]
+    Disabled,
+    WithTokenizer(TokenizerName),
+}
+
+impl Serialize for FastFieldOptions {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where S: serde::Serializer {
+        use serde::ser::SerializeMap;
+        match self {
+            FastFieldOptions::Enabled => serializer.serialize_bool(true),
```

Comment:
I appreciate the effort to have backward compatibliity, but we are breaking backward compat big time anyway in next release :D

We can probably avoid have something a tad more straightforward.

```rust


struct TextOption {
    fast: Option<TextFastFieldOption> //< disabled if None
}

#[derive(Serialize, Deserialize)]
struct TextFastFieldOption {
   tokenizer: Option<String>
}
```

Reply:
The compatibility is on a different layer, since `meta.json` is also used as input to create an index (e.g. `tantivy-cli`). Being able to parse existing `meta.json` files is quite helpful.

Reply:
This is adding unneeded complexity.
If it is not to interact with legacy stuff or with humans, let's avoid quirks like "this thing serializes as a bool or a string".

Reply:
I replaced it with a enum that can deserialize from existing `meta.json` without custom de/serialization